### PR TITLE
webgpu: Optimize concat/resizeBilinear/transpose

### PR DESCRIPTION
**Pull request by qjia7**
PERF

This changes increases work group size of concat, resizeBilinear, transpose.
Meanwhile, we only use x dimension for 'transpose' for better performance.

With this change, the micro-benchmarks have 3x~4x speedup.
                  pre     now
argMax (axis=0/1) 13ms -> 3ms
concat            15ms -> 5ms
resizeBilinear    17ms -> 5ms

Note: The speedup of argMax is caused by transpose.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/2115)

### CODE REVIEW

**Comment by qjia7** [comment]
@annxingyuan @kainino0x  Please take a look. Thanks.

File: tfjs-backend-webgpu/src/kernels/concat_webgpu.ts [hunk 1]
```diff
@@ -17,23 +17,25 @@
...
 import {WebGPUProgram} from './webgpu_program';
 
 export class ConcatProgram implements WebGPUProgram {
   outputShape: number[];
   userCode: string;
-  dispatchLayout: {x: number[], y: number[]};
+  dispatchLayout: {x: number[]};
   dispatch: [number, number, number];
   variableNames: string[];
+  workGroupSize: [number, number, number] = [64, 1, 1];
```
Comment (annxingyuan) [inline tfjs-backend-webgpu/src/kernels/concat_webgpu.ts]: Would you mind trying `workPerThread` > 1 to see whether that speeds things up even more?

File: tfjs-backend-webgpu/src/kernels/resize_bilinear_webgpu.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export class ResizeBilinearProgram implements WebGPUProgram {
   dispatchLayout: {x: number[], y: number[], z: number[]};
   dispatch: [number, number, number];
   variableNames = ['x'];
+  workGroupSize: [number, number, number] = [4, 4, 4];
```
Comment (annxingyuan) [inline tfjs-backend-webgpu/src/kernels/resize_bilinear_webgpu.ts]: Try `workPerThread` > 1.

File: tfjs-backend-webgpu/src/kernels/transpose_webgpu.ts [hunk 3]
```diff
@@ -27,6 +27,7 @@ export class TransposeProgram implements WebGPUProgram {
   dispatchLayout: {x: number[]};
   dispatch: [number, number, number];
   rank: number;
+  workGroupSize: [number, number, number] = [64, 1, 1];
```
Comment (annxingyuan) [inline tfjs-backend-webgpu/src/kernels/transpose_webgpu.ts]: Try `workPerThread` > 1.

File: tfjs-backend-webgpu/src/kernels/concat_webgpu.ts (hunk 1 above)
Comment (qjia7) [reply tfjs-backend-webgpu/src/kernels/concat_webgpu.ts]: Done. But I didn't see much differences between these two situations. See my comments in https://github.com/tensorflow/tfjs/pull/2118#issuecomment-539433429

File: tfjs-backend-webgpu/src/kernels/resize_bilinear_webgpu.ts (hunk 2 above)
Comment (qjia7) [reply tfjs-backend-webgpu/src/kernels/resize_bilinear_webgpu.ts]: When we use more than one dimension, we need to add a new helper function like getOutputCoordsWithWorkPerThread(int x, int y, int z). So I will do it in a separate patch.

File: tfjs-backend-webgpu/src/kernels/transpose_webgpu.ts (hunk 3 above)
Comment (qjia7) [reply tfjs-backend-webgpu/src/kernels/transpose_webgpu.ts]: Done

**Pull request merged**